src/osd/OSD.cc

```diff
@@ -7704,12 +7704,21 @@ vector<DaemonHealthMetric> OSD::get_health_metrics()
   {
     std::lock_guard l(pending_creates_lock);
     auto n_primaries = pending_creates_from_mon;
+    auto n_stuck_creating_pgs = 0;
+    const auto max_pgs_per_osd =
+             (cct->_conf.get_val<uint64_t>("mon_max_pg_per_osd") *
+             cct->_conf.get_val<double>("osd_max_pg_per_osd_hard_ratio"));
     for (const auto& create : pending_creates_from_osd) {
       if (create.second) {
 	n_primaries++;
       }
+      n_stuck_creating_pgs++;
+    }
+    if (n_stuck_creating_pgs && num_pgs >= max_pgs_per_osd ) {
```

Comment:
It seems to me that  `n_stuck_creating_pgs` may not be necessary. The check `num_pgs >= max_pgs_per_osd` appears sufficient to decide if the metric should be emplaced within the metrics vector.